distributed/scheduler.py

```diff
@@ -1818,14 +1832,17 @@ def transition_waiting_ready(self, key):
 
             if self.dependencies.get(key, None) or key in self.restrictions:
                 new_worker = decide_worker(self.dependencies, self.stacks,
-                        self.processing, self.who_has, self.has_what,
-                        self.restrictions, self.loose_restrictions, self.nbytes,
-                        key)
+                        self.stack_duration, self.processing, self.who_has,
+                        self.has_what, self.restrictions,
+                        self.loose_restrictions, self.nbytes, self.ncores, key)
                 if not new_worker:
                     self.unrunnable.add(key)
                     self.task_state[key] = 'no-worker'
                 else:
                     self.stacks[new_worker].append(key)
+                    duration = self.task_duration.get(key_split(key), 0.5)
```

Comment:
Yes, exponential weighted moving average based on calling `key_split` on the keyname.